x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureaistudio/AzureAiStudioServiceSettings.java

```diff
@@ -0,0 +1,140 @@
+            EnumSet.allOf(AzureAiStudioEndpointType.class),
+            validationException
+        );
+
+        AzureAiStudioProvider provider = extractRequiredEnum(
+            map,
+            PROVIDER_FIELD,
+            ModelConfigurations.SERVICE_SETTINGS,
+            AzureAiStudioProvider::fromString,
+            EnumSet.allOf(AzureAiStudioProvider.class),
+            validationException
+        );
+
+        return new BaseAzureAiStudioCommonFields(target, provider, endpointType, rateLimitSettings);
+    }
+
+    protected AzureAiStudioServiceSettings(StreamInput in) throws IOException {
+        this.target = in.readString();
+        this.provider = in.readEnum(AzureAiStudioProvider.class);
+        this.endpointType = in.readEnum(AzureAiStudioEndpointType.class);
+        if (in.getTransportVersion().onOrAfter(TransportVersions.ML_INFERENCE_RATE_LIMIT_SETTINGS_ADDED)) {
+            rateLimitSettings = new RateLimitSettings(in);
+        } else {
+            rateLimitSettings = DEFAULT_RATE_LIMIT_SETTINGS;
+        }
+    }
+
+    protected AzureAiStudioServiceSettings(
+        String target,
+        AzureAiStudioProvider provider,
+        AzureAiStudioEndpointType endpointType,
+        @Nullable RateLimitSettings rateLimitSettings
+    ) {
+        this.target = target;
+        this.provider = provider;
+        this.endpointType = endpointType;
+        this.rateLimitSettings = Objects.requireNonNullElse(rateLimitSettings, DEFAULT_RATE_LIMIT_SETTINGS);
+    }
+
+    protected record BaseAzureAiStudioCommonFields(
+        String target,
+        AzureAiStudioProvider provider,
+        AzureAiStudioEndpointType endpointType,
+        RateLimitSettings rateLimitSettings
+    ) {}
+
+    @Override
+    public DenseVectorFieldMapper.ElementType elementType() {
+        return DenseVectorFieldMapper.ElementType.FLOAT;
+    }
+
+    public String target() {
+        return this.target;
+    }
+
+    public AzureAiStudioProvider provider() {
+        return this.provider;
+    }
+
+    public AzureAiStudioEndpointType endpointType() {
+        return this.endpointType;
+    }
+
+    public RateLimitSettings rateLimitSettings() {
+        return this.rateLimitSettings;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(target);
+        out.writeEnum(provider);
+        out.writeEnum(endpointType);
+        rateLimitSettings.writeTo(out);
+    }
+
+    protected void addXContentFields(XContentBuilder builder, Params params) throws IOException {
+        this.addExposedXContentFields(builder, params);
+    }
+
+    protected void addExposedXContentFields(XContentBuilder builder, Params params) throws IOException {
+        builder.field(TARGET_FIELD, this.target);
+        builder.field(PROVIDER_FIELD, this.provider);
+        builder.field(ENDPOINT_TYPE_FIELD, this.endpointType);
+        rateLimitSettings.toXContent(builder, params);
```

Comment:
We actually want to hide this field for now until rate limiting is completely implemented here's an example: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/embeddings/OpenAiEmbeddingsServiceSettings.java#L264